File: src/main/java/ru/lapinlisss/olympic_api/service/AthleteService.java
package ru.lapinlisss.olympic_api.service;

import ru.lapinlisss.olympic_api.model.entity.Athlete;

import java.util.List;

public interface AthleteService {

    Athlete getAthleteById(Long id);

    List<Athlete> findAllAthletes(int page);

    List<Athlete> getAthletesByName(String name, String surname);

    List<Athlete> getAthletesByCountry(String country, int page);

    List<Athlete> getAthletesByGame(long id, int page);

    List<Athlete> getAthletesByGame(long id);

    List<Athlete> getAthletesByGameAndCountry(String type, int year, String country);

    Athlete putImageUrlForAthlete(long id, String url);
}

File: ui/olympic-ui/src/services/athlete-service.js
import ServiceUtil from "./service-util";

class AthleteService {

    getAllAthletes(page) {
        return ServiceUtil.call("get", null, `/athlete/all?page=${page}`);
    }

    getAthleteById(id) {
        return ServiceUtil.call("get", null, `/athlete/${id}`);
    }

    getAthleteByName(name, surname) {
        return ServiceUtil.call("get", null, `/athlete?name=${name}&surname=${surname}`);
    }

    getAthletesByCountry(country, page) {
        return ServiceUtil.call("get", null, `/athlete/country?counrty=${country}&page=${page}`)
    }

    getAthletesByGame(id, page) {
        return ServiceUtil.call("get", null, `/athlete/game?id=${id}&page=${page}`);
    }

    getTop50AthletesByGame(id) {
        return ServiceUtil.call("get", null, `/athlete/game/${id}/top50`);
    }

    getAthletesByGameAndCountry(type, year, country) {
        return ServiceUtil.call("get", null, `/athlete/game/country?type=${type}&year=${year}&country=${country}`)
    }

    postImgUrl(id, url) {
        return ServiceUtil.call("post", null, `/athlete/${id}/img?url=${url}`);
    }

}

export default new AthleteService;